solidity/contracts/SisypheanExchange.sol

```diff
@@ -199,10 +199,12 @@ contract SisypheanExchange is ForkedERC1155 {
 		Universe memory universe = universes[_universeId];
 		require(universe.forkingMarket == 0, "Universe is forked");
 		MarketResolutionData memory marketResolutionData = marketResolutions[_universeId][_marketId];
+		uint8 initialReportOutcome = marketResolutionData.outcome;
+		require(_outcome != initialReportOutcome, "Dispute must be for a different outcome than the currently winning one");
 		require(block.timestamp < marketResolutionData.reportTime + DISPUTE_PERIOD, "Market not in dispute window");
 		require(_outcome < 3, "Invalid outcome");
 
-		universe.reputationToken.transferFrom(msg.sender, address(this), REP_BOND * 2);
+		uint256 disputeStake = REP_BOND * 2;
```

Comment:
Possibly not a problem, but from a reentrancy standpoint I generally like moving the asset first, and then updating accounting after.  This change causes internal accounting to be updated first, and then the REP transferred which means we will need to pay closer attention during audits.

Is it necessary to move the REP after updating accounting here, or could we move the REP migration calls up here?

Reply:
It helps to do the accounting first since its no longer just a transfer but a migration which will check and use state for its logic. They also aren't really extracting anything in this function so not a typical reentrancy target, though yea its something we'd need to consider.